discovery_engine/test/repository/hive_engine_state_repo_test.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:typed_data' show Uint8List;
+
+import 'package:hive/hive.dart' show Box, Hive;
+import 'package:test/test.dart';
+
+import 'package:xayn_discovery_engine/src/infrastructure/box_name.dart'
+    show engineStateBox;
+import 'package:xayn_discovery_engine/src/infrastructure/repository/hive_engine_state_repo.dart'
+    show HiveEngineStateRepository;
+
+Future<void> main() async {
+  group('HiveEngineStateRepository', () {
+    late Box<Uint8List> _box;
+    final _repo = HiveEngineStateRepository();
```

Comment:
Shouldn't we create it fresh in each test?

Reply:
the repo itself in it's current form is stateless, so there is no need, but if we change it so that we open a box when creating an instance, then yes, we will need to update tests accordingly.

Reply:
Since is not a requirement that it must be state less we can do it so the tests don't have to assume that is stateless or stateful.

Reply:
If tests on repository are similar we can do this n the PR to open the box only once.

Reply:
I've adjusted other repo tests, other changes might be better to do in a separate PR